src/core/device.c

```diff
@@ -114,12 +159,134 @@ static void device_done(Unit *u) {
         assert(d);
 
         device_unset_sysfs(d);
+        free(d->devname);
         d->wants_property = strv_free(d->wants_property);
 }
 
+static int root_cgroup_apply_block_device_io_cost_qos(Unit *root_slice_unit, Device *d) {
+        char buf[DECIMAL_STR_MAX(dev_t)*2+9+(6+DECIMAL_STR_MAX(uint64_t)+1)*12];
+        IOCostQos *io_cost_qos;
+        dev_t dev;
+        int r;
+
+        assert(root_slice_unit);
+        assert(d);
+
+        io_cost_qos = &d->io_cost_qos;
+
+        /* io_cost_qos enabled < 0 indicates that user did not set the enabled field */
+        if (!io_cost_qos || io_cost_qos->enabled < 0)
```

Comment:
`io_cost_qos` is non-null since you took the address above. So `!io_cost_qos` isn't really necessary